tpython/lib/utils/futils.py

```diff
@@ -1,136 +1,129 @@
+        sh.copy2(src, dest)
+
 
 def copyFolder(src, dest):
-    os.mkdir('-p', dest)
-    sh.copy('-rf', src, dest)
+    sh.copytree(src, dest)
 
-def readWholeFile(path1):
-    if os.path.exists(path1):
-        array = os.open(path1)
-        return array
+
+def readWholeFile(path):
+    if os.path.exists(path):
+        open_content = open(path)
+        file_content = open_content.read()
+        return file_content
     else:
-        logging.warning('Unable to locate ' + path1)
+        print("Unable to locate " + path)
+
 
 def readYaml(path):
-    yamldata = ''
-    try:
-        yamldata = yaml.load(os.open(path, encoding='utf-8'))
-    except:
-        logging.warning('Tutors ${version} encountered an error reading properties.yaml:')
-        logging.warning('--------------------------------------------------------------')
-        #logging.warning(err.mark.buffer)
-        logging.warning('--------------------------------------------------------------')
-        #logging.warning(err.message)
-        logging.warning('Review this file and try again....')
+    yamldata = ""
+    with open(path) as yam:
+        yamldata = frontmatter.load(yam)
     return yamldata
 
+
 def readEnrollment(path):
-    yamldata = ''
-    try:
-        yamldata = yaml.load(os.open(path, encoding='utf-8'))
-    except:
-        logging.warning('Tutors ${version} encountered an error reading the enrollment file:')
-        logging.warning('--------------------------------------------------------------')
-        #logging.warning(err.mark.buffer)
-        logging.warning('--------------------------------------------------------------')
-        #logging.warning(err.message)
-        logging.warning('Ignoring enrolling file for the moment....')
+    yamldata = ""
+    with open(path) as yam:
+        yamldata = frontmatter.load(yam)
     return yamldata
 
+
 def readCalendar(path):
-    yamldata = ''
-    try:
-        yamldata = yaml.load(os.open(path, encoding='utf-8'))
-    except:
-        logging.warning('Tutors ${version} encountered an error reading the calendar file:')
-        logging.warning('--------------------------------------------------------------')
-        #logging.warning(err.mark.buffer)
-        logging.warning('--------------------------------------------------------------')
-        #logging.warning(err.message)
-        logging.warning('Ignoring calendar file for the moment....')
+    yamldata = ""
+    with open(path) as yam:
+        yamldata = frontmatter.load(yam)
     return yamldata
 
 
-def getHeader(fileName):
-    header = ''
-    array = os.read(fileName).split('\n')
-    if array[0][0] == '#':
-        header = array[0][0:2]
-    else:
-        header = array[0]
-    return header
-
 def withoutHeader(fileName):
-    content = os.read(fileName)
-    line1 = content.index('\n')
-    content = content[line1 + 1 : len(content)]    
+    file = open(fileName, "r")
+    content = file.read()
+    line1 = content.index("\n")
```

Comment:
it might be better to use some built-in functions instead of manually relying on the newline character. 

You could use something like https://docs.python.org/3/library/io.html#io.IOBase.readline or even https://docs.python.org/3/library/io.html#io.IOBase.readlines

You could even just use `for line in file:` which will iterate through each line for you.

Reply:
This is Eamonns logic, I've tried changing it but can't seem to get the right result another way. I'm only using the newline char to get the index of the end of the 1st line which is then used as a sub-string value.

I'm not sure if i'm missing the point here?